Foundation/NSPathUtilities.swift

```diff
@@ -550,13 +564,61 @@ extension NSString {
     }
     
     public func getFileSystemRepresentation(_ cname: UnsafeMutablePointer<Int8>, maxLength max: Int) -> Bool {
+#if os(Windows)
+        let fsr = UnsafeMutablePointer<WCHAR>.allocate(capacity: max)
+        defer{ fsr.deallocate() }
+        guard _getFileSystemRepresentation(fsr, maxLength: max) else { return false }
+        let str = String(decodingCString: fsr, as: UTF16.self)
+        return str.withCString() {
+            let chars = strnlen_s($0, max)
+            guard chars < max else {
+                return false
+            }
+            cname.assign(from: $0, count: chars + 1)
+            return true
+        }
+#else
         guard self.length > 0 else {
             return false
         }
-        
         return CFStringGetFileSystemRepresentation(self._cfObject, cname, max)
+#endif
     }
 
+#if os(Windows)
+    public func _getFileSystemRepresentation(_ cname: UnsafeMutablePointer<foundation_char_t>, maxLength max: Int) -> Bool {
+        guard self.length > 0 else {
```

Comment:
This is probably what I'm most concerned about, I'm not entirely sure my logic is 100% sound. `CreateFileW` is pretty good about being able to figure out a path, so in general, we just have to get the path to be "good enough" to be parsed by it, and it will properly do the logic for the various types of Windows paths. However, there a few exceptions I've found:
- /c:/foo/bar doesn't parse, the first '/' needs to be removed
  - except `/foo/bar` should parse, since it's a drive relative absolute path
- Trailing slashes must be removed
  - except in the case of `[driveletter]:\` where `C:\` is the `C` root directory, but `C:` is the current directory.